core/src/main/java/org/apache/iceberg/StaticTable.java

```diff
@@ -0,0 +1,262 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import org.apache.iceberg.encryption.EncryptionManager;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.io.LocationProvider;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+
+// TODO: any ideas on naming?
+public class StaticTable implements Table, Serializable {
+  private final String name;
+  private final String location;
+  private final Map<String, String> properties;
+  private final Schema schema;
+  private final PartitionSpec spec;
+  private final Map<Integer, PartitionSpec> specs;
+  private final SortOrder sortOrder;
+  private final Map<Integer, SortOrder> sortOrders;
+  private FileIO io;
```

Comment:
These are not final so that we can override them using setters.

Reply:
Makes sense for `FileIO`, but do we need to do this for `EncryptionManager` and `LocationProvider`?

Reply:
No, can be just `io` for now.

Reply:
No longer applies.